lib/ssl/src/tls_connection.erl

```diff
@@ -662,8 +672,18 @@ hello(internal, #server_hello{extensions = Extensions} = Hello,
       #state{ssl_options = #{handshake := hello},
              handshake_env = HsEnv,
              start_or_recv_from = From} = State) ->
-    {next_state, user_hello, State#state{start_or_recv_from = undefined,
-                                         handshake_env = HsEnv#handshake_env{hello = Hello}},
+    %% RFC6066.8, If a server returns a "CertificateStatus" message,
+    %% then the server MUST have included an extension of type
+    %% "status_request" with empty "extension_data" in the extended
+    %% server hello.
+    OcspState = HsEnv#handshake_env.ocsp_stapling_state,
+    IsCertStatusRequested = maps:is_key(ocsp_stapling, Extensions),
+    {next_state, user_hello,
+     State#state{start_or_recv_from = undefined,
+                 handshake_env = HsEnv#handshake_env{
+                     hello = Hello,
+                     ocsp_stapling_state = OcspState#{
+                         is_server_hello_requested := IsCertStatusRequested}}},
```

Comment:
We can set that above mentioned single variable here to the negotiated value.

Reply:
> We can set that above mentioned single variable here to the negotiated value.

done